src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -303,6 +304,25 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			if (contents) {
 				this._defaultLanguageInfo = contents.metadata && contents.metadata.language_info;
 				this._savedKernelInfo = this.getSavedKernelInfo(contents);
+				if (contents.metadata) {
+					//Telemetry of loading notebook
+					if (contents.metadata.azdata_notebook_guid && contents.metadata.azdata_notebook_guid.length === 36) {
+						//Verify if it is actual GUID and then send it to the telemetry
+						let regex = new RegExp('(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}');
+						if (regex.test(contents.metadata.azdata_notebook_guid)) {
```

Comment:
You need to match the beginning and end as well - this currently accepts any string as long as it  has a GUID somewhere in it

Reply:
It wont accept any string as `contents.metadata.azdata_notebook_guid.length === 36 `limits the length of the string to guid length.

Reply:
It wont accept any string as `contents.metadata.azdata_notebook_guid.length === 36 `limits the length of the string to guid length.

Reply:
Hm, true but I don't know why you wouldn't just use the beginning/end markers regardless. You're already doing a regex so might as well consolidate the whole check into the one test.

Reply:
I feel its redundant as it does not add any extra validation. length check will restrict the size, regex will validate the resulting string is a guid. Not sure how adding beginning/end markers check will help here. The current code will not send anything other than a guid.